web-console/src/lib/components/connectors/dialogs/tabs/GenericConnectorForm.tsx

```diff
@@ -0,0 +1,178 @@
+          aria-describedby='validation-name'
+          disabled={props.disabled}
+        />
+      </Grid>
+      <Grid item sm={8} xs={12}>
+        <TextField
+          name='description'
+          label='Description'
+          size='small'
+          fullWidth
+          id='connector-description' // referenced by webui-tester
+          placeholder={PLACEHOLDER_VALUES['connector_description']}
+          aria-describedby='validation-description'
+          disabled={props.disabled}
+        />
+      </Grid>
+      <Grid item sm={12} xs={12}>
+        <Box sx={{ height: { xs: '50vh', md: '40vh' } }}>
+          <JSONConfigEditor {...props}></JSONConfigEditor>
+        </Box>
+      </Grid>
+    </Grid>
+  )
+}
+
+const JSONConfigEditor = (props: {
+  disabled?: boolean
+  configFromText: (t: string) => unknown
+  configToText: (c: unknown) => string
+}) => {
+  const theme = useTheme()
+  const vscodeTheme = theme.palette.mode === 'dark' ? 'vs-dark' : 'vs'
+  const ctx = useFormContext()
+  const configText: string = ctx.watch('config')
+  const { errors } = useFormState({ control: ctx.control })
+  const editorRef = useRef<editor.IStandaloneCodeEditor>()
+  const monaco = useMonaco()!
+  useEffect(() => {
+    if (!editorRef.current) {
+      return
+    }
+    const errorMarkers = Object.entries(errors?.config ?? {}).map(([field, error]) => {
+      const offenderPos = (model =>
+        model.findNextMatch(`"${field}":`, { lineNumber: 0, column: 0 }, false, true, null, false) ??
+        model.findNextMatch(
+          `"${field.replaceAll('_', '.')}":`,
+          { lineNumber: 0, column: 0 },
+          false,
+          true,
+          null,
+          false
+        ))(editorRef.current!.getModel()!)?.range
+
+      const defaultErr = {
+        startLineNumber: 0,
+        endLineNumber: 0,
+        startColumn: 0,
+        endColumn: 1,
+        message: error.message,
+        severity: monaco.MarkerSeverity.Error
+      }
+      if (!offenderPos) {
+        return defaultErr
+      }
+      return {
+        startLineNumber: offenderPos.startLineNumber,
+        endLineNumber: offenderPos.endLineNumber,
+        startColumn: offenderPos.startColumn,
+        endColumn: offenderPos.endColumn,
+        message: error.message,
+        severity: monaco.MarkerSeverity.Error
+      }
+    })
+    monaco.editor.setModelMarkers(editorRef.current.getModel()!, 'config-errors', errorMarkers)
+  }, [errors, editorRef])
+  function handleEditorDidMount(editor: editor.IStandaloneCodeEditor, monaco: Monaco) {
+    editorRef.current = editor
+    // Only process input when we have finished typing
+    editor.onDidBlurEditorText(() => {
+      try {
+        const v = props.configFromText(editor.getValue() ?? '')
+        ctx.setValue('config', v)
+        monaco.editor.setModelMarkers(editorRef.current!.getModel()!, 'parse-errors', [])
+      } catch (e) {
+        invariant(e instanceof Error)
```

Comment:
so there are no libraries doing this already?

Reply:
For generating error markers from errors? Doesn't seem to be. I will create helper functions when there are more instances of similar code to generalize